test/optimism/TokenRateNotifier.unit.test.ts

```diff
@@ -0,0 +1,126 @@
+import hre from "hardhat";
+import { assert } from "chai";
+import { unit } from "../../utils/testing";
+import { BigNumber, utils } from 'ethers'
+import { ethers } from 'hardhat'
+
+import {
+    TokenRateNotifier__factory,
+    ObserversArray__factory,
+    OpStackTokenRateObserver__factory,
+    ITokenRateObserver__factory,
+} from "../../typechain";
+
+unit("TokenRateNotifier", ctxFactory)
```

Comment:
if we decide to separate `ObservsArray` a dedicated suit is required to maintain reusability of `ObserversArray`

Reply:
not valid anymore